realm/realm-library/src/main/java/io/realm/internal/async/RealmThreadPoolExecutor.java

```diff
@@ -55,6 +59,42 @@ public static RealmThreadPoolExecutor newSingleThreadExecutor() {
         return new RealmThreadPoolExecutor(1, 1);
     }
 
+    /**
+     * Try using the number of files named 'cpuNN' in sysfs to figure out the number of
+     * processors on this device. `Runtime.getRuntime().availableProcessors()` may return
+     * a smaller number when the device is sleeping.
+     *
+     * @return the number of threads to be allocated for the executor pool
+     */
+    @SuppressWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
```

Comment:
Sure.  Could do that.  I think this is pretty much equivalent, and it doesn't require a new `provided` library.